debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java

```diff
@@ -568,67 +548,36 @@ static void logError(TransactionalBufferMetrics metrics, String format, Object..
      * @return size
      */
     private static int getRedoLogGroupSize(OracleConnection connection) throws SQLException {
-        return getMap(connection, SqlUtils.allOnlineLogsQuery(), getDatabaseMaxScnValue(connection)).size();
-    }
-
-    /**
-     * Get the database maximum SCN value
-     *
-     * @param connection the oracle database connection
-     * @return the maximum scn value as a string value
-     */
-    public static String getDatabaseMaxScnValue(OracleConnection connection) {
-        OracleDatabaseVersion version = connection.getOracleVersion();
-        if ((version.getMajor() == 19 && version.getMaintenance() >= 6) || (version.getMajor() > 19)) {
-            return MAX_SCN_19_6;
-        }
-        else if ((version.getMajor() == 12 && version.getMaintenance() >= 2) || (version.getMajor() > 12)) {
-            return MAX_SCN_12_2;
-        }
-        else if ((version.getMajor() == 11 && version.getMaintenance() >= 2) || (version.getMajor() == 12 && version.getMaintenance() < 2)) {
-            return MAX_SCN_11_2;
-        }
-        throw new RuntimeException("Max SCN cannot be resolved for database version " + version);
+        return connection.queryAndMap("SELECT COUNT(DISTINCT GROUP#) FROM V$LOG", (rs) -> rs.getInt(1));
     }
 
     /**
      * This method returns all online log files, starting from one which contains offset SCN and ending with one containing largest SCN
      * 18446744073709551615 on Ora 19c is the max value of the nextScn in the current redo todo replace all Long with BigInteger for SCN
      */
-    public static Map<String, BigInteger> getOnlineLogFilesForOffsetScn(OracleConnection connection, Long offsetScn) throws SQLException {
-        // TODO: Make offset a BigInteger and refactor upstream
-        BigInteger offsetScnBi = BigInteger.valueOf(offsetScn);
-        LOGGER.trace("Getting online redo logs for offset scn {}", offsetScnBi);
-        Map<String, ScnRange> redoLogFiles = new LinkedHashMap<>();
-
-        String maxScnStr = getDatabaseMaxScnValue(connection);
-        final BigInteger maxScn = new BigInteger(maxScnStr);
+    public static Map<String, Scn> getOnlineLogFilesForOffsetScn(OracleConnection connection, Scn offsetScn) throws SQLException {
+        LOGGER.trace("Getting online redo logs for offset scn {}", offsetScn);
+        Map<String, Scn> redoLogFiles = new LinkedHashMap<>();
```

Comment:
Should there be a dedicated domain object encapsulating file name and SCN? Like `LogFile` or so? I feel that'd make sense :)

Reply:
Seems reasonable.